ts/private/ts_project_worker.js

```diff
@@ -243,10 +244,22 @@ function createProgram(args, initialInputs) {
             return { close: () => {} };
         }
         debuglog(`watchDirectory ${directoryPath}`);
-        directoryWatchers.set(directoryPath, (input) => callback(path.join(execRoot, input)));
+
+        const cb = (input) => callback(path.join(execRoot, input))
+
+        const callbacks = directoryWatchers.get(directoryPath) || new Set();
+        callbacks.add(cb)
+        directoryWatchers.set(directoryPath, callbacks);
+
         return {
             close: () => {
-                directoryWatchers.delete(directoryPath);
+                const callbacks = directoryWatchers.get(directoryPath);
+                callbacks.delete(cb);
+                if (callbacks.size == 0) {
```

Comment:
Should be `===`. Basically never use `==`, it's essentially a mistake from the 90s that can never be undone :(